### DISCUSSION TITLE
GitLab job examples?

### DISCUSSION THREAD

**Discussion by rfulwell** [devcontainers/spec#147] [Ideas]
I see the [GitHub Action and Azure DevOps Task](https://github.com/devcontainers/ci). It should be straightforward to implement the dev container build as a GitLab CI job. Is there an existing example of this?

**Comment by Chuxel** [comment]
You should be able to use the Dev Container CLI directly assuming Docker is installed on the host the job is run on.  

https://containers.dev/implementors/reference/#what-is-CLI

For examples:
1. This is an example script that will pre-build an image: https://github.com/devcontainers/cli/blob/main/example-usage/image-build/build-image.sh
2. And this is an example of using a dev container for your build: https://github.com/devcontainers/cli/blob/main/example-usage/ci-app-build-script/build-application.sh

We could add cross-references to these in docs as I think about it.
